porch/repository/pkg/cache/repository.go

```diff
@@ -119,8 +120,8 @@ func (r *cachedRepository) getPackages(ctx context.Context, forceRefresh bool) (
 
 	if packages == nil {
 		// TODO: Avoid simultaneous fetches?
-		var p []repository.PackageRevision
-		p, err = r.repo.ListPackageRevisions(ctx)
+		// TODO: Push-down partial refresh?
```

Comment:
what is meant by "push-down"?

Reply:
Ah, by push-down I mean pushing the predicates down into the lower levels, so that they can also optimize their fetching to just return the filtered results.  It's not obviously a win here, because if we push-down the filter, then we can't cache the results (well, not without _much_ more complex caching!)